### ISSUE TITLE
bug: CircularModuleDependency only detects 2-node cycles, misses A→B→C→A

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#22]
## Description

`CircularModuleDependency` rule only detects self-references and direct A↔B cycles. Multi-node cycles (A→B→C→A, A→B→C→D→A, etc.) are silently ignored.

## Root Cause

The `checkForCycles()` method (line 110-128) only checks if a direct dependency has a back-edge to the current module:

```kotlin
module.dependencies.forEach { depName ->
    val dependency = modules.find { it.name == depName }
    if (dependency != null && dependency.dependencies.contains(module.name)) {
        // Only detects direct back-edge
    }
}
```

## Reproduction

```kotlin
val moduleA = module {
    includes(moduleB)
}

val moduleB = module {
    includes(moduleC)
}

val moduleC = module {
    includes(moduleA)  // Creates A→B→C→A cycle - NOT DETECTED!
}
```

**Expected:** Rule reports circular dependency  
**Actual:** No finding reported

## Note

The rule also only operates within a single file — cross-file circular dependencies are a known limitation of detekt's per-file analysis.
